Source/CombatExtended/CombatExtended/Verbs/Verb_ShootCE.cs

```diff
@@ -94,10 +94,42 @@ protected override float SwayAmplitude
         // Whether our shooter is currently under suppressive fire
         private bool IsSuppressed => ShooterPawn?.TryGetComp<CompSuppressable>()?.isSuppressed ?? false;
 
+        //Checks if magazine-fed weapon needs to reload before able to shoot again (out-of-ammo state is automatically handled by TryStartReload)
+        private bool NeedsMagazineReload => CompAmmo != null && CompAmmo.HasMagazine && CompAmmo.CurMagCount == 0;
+
+        //Mag-less weapons need separate handling for out-of-ammo state, as TryStartReload does nothing for them
+        private bool IsMaglessWeaponOutOfAmmo => CompAmmo != null && !CompAmmo.HasMagazine && CompAmmo.UseAmmo && !CompAmmo.HasAmmo;
+
+        private bool IsAttacking => ShooterPawn?.CurJobDef == JobDefOf.AttackStatic || ShooterPawn?.stances.curStance is Stance_Warmup;
+
         #endregion
 
         #region Methods
 
+        public override bool Available()
```

Comment:
This should be handled via the base class

Reply:
I put this in Verb_ShootCE because there's also an added reload check at the end of TryCastShot(), and it made sense to me that all the reload logic should be contained in the same class. With that in mind, should the TryCastShot() reload check also be moved into LaunchProjectile?

Reply:
The base class already has a reload check though.